pkg/subctl/cmd/validate_tunnel.go

```diff
@@ -0,0 +1,186 @@
+/*
+© 2021 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	subv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+
+	"github.com/submariner-io/submariner-operator/apis/submariner/v1alpha1"
+)
+
+const (
+	ClientSourcePort = "9898"
```

Comment:
Ok, I think this can be problematic. If we filter on source ports to detect that a packet arrives properly to the gateway, that can fail because the SRC port under some nat conditions can be modified.

As we were talking in slack, may be we can follow the next approach:

1) Try to find packets with src "ClientSourcePort", if that fails, push a WARNING about this
2) Try to find packets with the specific content (tcpdump matching on specific packet data), if this fails ERROR.

sounds ok? :)

Reply:
Yup. I'll update the PR.